app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4333,15 +4423,40 @@ private void setupVideoSurface() {
     }
 
     private void cleanupVideoSurface() {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) { // >=API23
-            if (surfaceHolderCallback != null) {
-                if (binding != null) {
-                    binding.surfaceView.getHolder().removeCallback(surfaceHolderCallback);
-                }
-                surfaceHolderCallback.release();
-                surfaceHolderCallback = null;
+        // Only for API >= 23
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && surfaceHolderCallback != null) {
+            if (binding != null) {
+                binding.surfaceView.getHolder().removeCallback(surfaceHolderCallback);
             }
+            surfaceHolderCallback.release();
+            surfaceHolderCallback = null;
         }
     }
     //endregion
+
+    /**
+     * Get the video renderer index of the current playing stream.
+     *
+     * This method returns the video renderer index of the current
+     * {@link MappingTrackSelector.MappedTrackInfo} or {@link #RENDERER_UNAVAILABLE} if the current
+     * {@link MappingTrackSelector.MappedTrackInfo} is null or if there is no video renderer index.
+     *
+     * @return the video renderer index or {@link #RENDERER_UNAVAILABLE} if it cannot be get
+     */
+    private int getVideoRendererIndex() {
+        final MappingTrackSelector.MappedTrackInfo mappedTrackInfo = trackSelector
+                .getCurrentMappedTrackInfo();
+
+        if (mappedTrackInfo != null) {
+            for (int i = 0; i < mappedTrackInfo.getRendererCount(); i++) {
+                final TrackGroupArray trackGroups = mappedTrackInfo.getTrackGroups(i);
+                if (!trackGroups.isEmpty()
+                        && simpleExoPlayer.getRendererType(i) == C.TRACK_TYPE_VIDEO) {
+                    return i;
+                }
+            }
```

Comment:
I think the Java Streaming API would be great here - less code, more understabale, no additioanl variables and for loops.
```java
return Instream.range(0, mappedTrackInfo.getRendererCount()) // Check every renderer
  .findFirst(i ->!mappedTrackInfo.getTrackGroups(i).isEmpty() // IDK Check where the trackgroup has something 
                  && simpleExoPlayer.getRendererType(i) == C.TRACK_TYPE_VIDEO) // Check where the renderer is a video renderer
 .orElse(RENDERER_UNAVAILABLE); // Otherwise return default
```